.github/workflows/terraform-test-pipeline.yml

```diff
@@ -90,45 +94,153 @@ jobs:
+                    # sanitize key
+                    key=${env_var_key//-}  # remove '-'
+                    key=${key// }          # remove space
+                    # Get the corresponding value
+                    env_var_value=$(yq e ".CRA_TARGETS[0].CRA_ENVIRONMENT_VARIABLES.${key}" "${{ inputs.craConfigYamlFile }}")
+
+                    echo "Exporting environment variable: ${key}=${env_var_value}"
+                    export "${key}"="${env_var_value}"
+                  done <<< "$env_var_keys_output"
+                fi
+
+
+                # Run CRA scan for the target
+                if [[ -n "${CRA_TARGET}" ]]; then
+                  echo "CRA IGNORE FILE: ${CRA_IGNORE_RULES_FILE}"
+                  ci/run-cra-scc-v2.sh "$CRA_TARGET"
+                else
+                  echo "No CRA target specified for this configuration!"
+                fi
+              done
 
-          # Run CRA scan for each target in the comma-separated list
-          IFS=',' read -ra CRA_TARGETS_ARRAY <<< "${{ inputs.craTarget }}"
-          for target in "${CRA_TARGETS_ARRAY[@]}"; do
-            if ${{ inputs.craSCCv2 }}; then
-              echo "CRA IGNORE FILE: ${CRA_IGNORE_RULES_FILE}"
-              ci/run-cra-scc-v2.sh "$target"
+              # Unset the variables
+              # Iterate over each key
+              while IFS= read -r env_var_key; do
+                # sanitize key
+                key=${env_var_key//-}  # remove '-'
+                key=${key// }          # remove space
+
+                echo "Unset environment variable: ${key}"
+                unset "${key}"
+              done <<< "$env_var_keys_output"
+              if [[ -n "${CRA_IGNORE_RULES_FILE}" ]]; then
+                  unset CRA_IGNORE_RULES_FILE
+              fi
+            else
+              echo "Provided YAML configuration file '${{ inputs.craConfigYamlFile }}' not found!"
+              exit 1
+            fi
+          else
+            export CRA_IGNORE_GOALS_FILE="$(pwd)/${{ inputs.craGoalIgnoreFile }}"
+            export CRA_IGNORE_RULES_FILE="$(pwd)/${{ inputs.craRuleIgnoreFile }}"
+
+            ENV_VARS="${{ inputs.craEnvironmentVariables }}"
+            if [[ -n "${ENV_VARS}" ]]; then
+              IFS=',' read -ra ENV_VARS_ARRAY <<< "$ENV_VARS"
+              for i in "${ENV_VARS_ARRAY[@]}"; do
+                # Use an array to store the variable and value
+                ENV_VAR=(${i//=/ })
+                echo "Exporting environment variable: ${ENV_VAR[0]}"
+                export "${ENV_VAR[0]}"="${ENV_VAR[1]}"
+              done
             else
-              echo "CRA IGNORE FILE: ${CRA_IGNORE_GOALS_FILE}"
-              ci/run-cra.sh "$target"
+              echo "No input environment variables to export"
+            fi
+
+            # Install CRA plugin if not detected
+            if ! ibmcloud plugin show cra; then
+              ibmcloud plugin install cra -f
```

Comment:
Looks like this install of the plugin will not occur if a yaml config is specified? It looks like a precautionary step, but you could move the logic to very top of script to make sure it always runs.